From 31010e62422ffb231fa5f8de8833750e40c2a1d4 Mon Sep 17 00:00:00 2001
From: Teo Hall <teo.hall@nxp.com>
Date: Thu, 8 Feb 2018 15:37:40 -0600
Subject: [PATCH 3358/5242] MLK-17570: Add i.MX8QP support

commit  a24e4010468b4fd5a3ab69c1ef5450010ca442c5 from
https://source.codeaurora.org/external/imx/linux-imx.git

Add Kconfig and DTS support for i.MX8QP with a simplified structure

Signed-off-by: Teo Hall <teo.hall@nxp.com>

Squashed into a single commit during 4.14 rebase.

Signed-off-by: Leonard Crestez <leonard.crestez@nxp.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 arch/arm64/Kconfig                                 |    2 +-
 arch/arm64/Kconfig.platforms                       |   11 ++++++
 arch/arm64/boot/dts/freescale/Makefile             |    1 +
 .../boot/dts/freescale/fsl-imx8qp-lpddr4-arm2.dts  |   41 ++++++++++++++++++++
 4 files changed, 54 insertions(+), 1 deletion(-)
 create mode 100644 arch/arm64/boot/dts/freescale/fsl-imx8qp-lpddr4-arm2.dts

diff --git a/arch/arm64/Kconfig b/arch/arm64/Kconfig
index 41ec0bb..0023e9b 100644
--- a/arch/arm64/Kconfig
+++ b/arch/arm64/Kconfig
@@ -863,7 +863,7 @@ config FORCE_MAX_ZONEORDER
 	int
 	default "14" if (ARM64_64K_PAGES && TRANSPARENT_HUGEPAGE)
 	default "12" if (ARM64_16K_PAGES && TRANSPARENT_HUGEPAGE)
-	default "14" if (ARCH_FSL_IMX8QM || ARCH_FSL_IMX8QXP || ARCH_FSL_IMX8MQ)
+	default "14" if (ARCH_FSL_IMX8QM || ARCH_FSL_IMX8QXP || ARCH_FSL_IMX8MQ || ARCH_FSL_IMX8QP)
 	default "11"
 	help
 	  The kernel memory allocator divides physically contiguous memory
diff --git a/arch/arm64/Kconfig.platforms b/arch/arm64/Kconfig.platforms
index 58f52b2..e097762 100644
--- a/arch/arm64/Kconfig.platforms
+++ b/arch/arm64/Kconfig.platforms
@@ -306,6 +306,17 @@ config ARCH_FSL_IMX8QM
 	help
 	  This enables support for Freescale i.MX8QM SOC.
 
+config ARCH_FSL_IMX8QP
+	bool "Freescale i.MX8QP SOC"
+	select PINCTRL
+	select PINCTRL_IMX
+	select PINCTRL_IMX8QM
+	select PM
+	select SOC_BUS
+	help
+	  This enables support for Freescale i.MX8QP SOC.
+
+
 config ARCH_FSL_IMX8QXP
 	bool "Freescale i.MX8QXP SOC"
 	select PINCTRL
diff --git a/arch/arm64/boot/dts/freescale/Makefile b/arch/arm64/boot/dts/freescale/Makefile
index 0b24ded3..cb04fd5 100644
--- a/arch/arm64/boot/dts/freescale/Makefile
+++ b/arch/arm64/boot/dts/freescale/Makefile
@@ -36,6 +36,7 @@ dtb-$(CONFIG_ARCH_FSL_IMX8QM) += fsl-imx8qm-lpddr4-arm2.dtb \
 				 fsl-imx8qm-lpddr4-arm2-dsi-rm67191.dtb \
 				 fsl-imx8qm-lpddr4-arm2-enet2-tja1100.dtb \
 				 fsl-imx8qm-lpddr4-arm2-hsic.dtb
+dtb-$(CONFIG_ARCH_FSL_IMX8QP) += fsl-imx8qp-lpddr4-arm2.dtb
 dtb-$(CONFIG_ARCH_FSL_IMX8QXP) += fsl-imx8qxp-lpddr4-arm2.dtb \
 				  fsl-imx8qxp-mek.dtb \
 				  fsl-imx8qxp-mek-dom0.dtb \
diff --git a/arch/arm64/boot/dts/freescale/fsl-imx8qp-lpddr4-arm2.dts b/arch/arm64/boot/dts/freescale/fsl-imx8qp-lpddr4-arm2.dts
new file mode 100644
index 0000000..00446b8
--- /dev/null
+++ b/arch/arm64/boot/dts/freescale/fsl-imx8qp-lpddr4-arm2.dts
@@ -0,0 +1,41 @@
+/*
+ * Copyright (C) 2016 Freescale Semiconductor, Inc.
+ * Copyright 2018 NXP
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU General Public License
+ * as published by the Free Software Foundation; either version 2
+ * of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ */
+
+#include "fsl-imx8qm-lpddr4-arm2.dts"
+
+/ {
+        model = "Freescale i.MX8QP ARM2";
+        compatible = "fsl,imx8qm-arm2", "fsl,imx8qm";
+};
+
+&gpu_3d0 {
+      assigned-clock-rates = <625000000>, <625000000>;
+};
+
+&gpu_3d1 {
+      assigned-clock-rates = <625000000>, <625000000>;
+};
+
+&A72_1 {
+        device_type = "";
+};
+
+&imx8_gpu_ss {/*<freq-kHz vol-uV>*/
+	operating-points = <
+/*nominal*/	625000	0
+		625000	0
+/*underdrive*/	400000	0  /*core/shader clock share the same frequency on underdrive mode*/
+	>;
+};
-- 
1.7.9.5

